frontend/src/components/advanced-auto-filter/advanced-auto-filter.tsx

```diff
@@ -112,8 +112,10 @@ const AdvancedAutoFilter: FC = () => {
       objectToQueryString({
         ...rangeFiltersToObject(rangeFilters),
         ...checkLists,
-        brandId: brandDetails.map((item) => item.brandId),
-        modelId: brandDetails.flatMap((item) => item.modelIds),
+        brandDetails: brandDetails.map(({ modelIds, brandId }) => ({
+          modelIds,
+          brandId,
+        })),
```

Comment:
key `brandDetails`  doesn't exist on [specification](https://docs.google.com/document/d/16dU-biODvptcGAsv2I6EBGBKiQkJLGKvraUJYM4CkcU/edit) rules. I think you can check brand details in filters and: if in brand details only `brandId `- pass `brandId` in query, if also present `modelIds` - pass only `modelIds `without their `brandId`. I think it should work, won't break spec. rules and could be simple to handle on the back-end side. Because previously I passed in query `brandId ` and its `modelIds` together. So my suggestion is to pass only `modelIds`, without their `brandId` (if it's possible of course)